### ISSUE TITLE
Default timeout in instruments.rb causes failure on slow hardware

### ISSUE THREAD

**Issue by gdknutel** [calabash/run_loop#259]
Hi,

I'm trying to run tests with cucumber-ios. Cucumber will fail to start sometimes when running on a real device, and almost always on the simulator. I have tracked this down to the version function in instruments.rb. No timeout is specified, which means the default value of 10 seconds is used in xcrun.exec. Since I am running on a fairly old Mac with a HDD, 10 seconds is not enough to run the command "xcrun instruments" to read the version number.

Increasing the default timeout to 30 seconds solved the issue for me, but it would be better if one could specify the timeout on the command line.

I am using run_loop-1.5.4, but I see the code is the same as of https://github.com/calabash/run_loop/blob/652ebecef5ee722d5066688e6bcd8a657255d1a7/lib/run_loop/instruments.rb#L114

Lines 109-118:
```ruby
# Returns the instruments version.
# @return [RunLoop::Version] A version object.
def version
  @instruments_version ||= lambda do
    args = ['instruments']
    hash = xcrun.exec(args, log_cmd: true)                 #No timeout specified here!
    version_str = hash[:err][VERSION_REGEX, 0]
    RunLoop::Version.new(version_str)
  end.call
end
```

**Comment by jmoody** [comment]
@gdknutel 

Ouch.  That is slow.  

> Increasing the default timeout to 30 seconds solved the issue for me,

We are trying to move away from setting command line variables for every timeout.  They are hard to document, deprecate, and manage.

I'll see what I can do for the next version.

**Comment by jmoody** [comment]
I have bumped the timeout to 30 seconds.

I will not add an environment variable for this unless absolutely necessary.

More interesting to me is why `Instruments#version` is being called.  Any way you can get a stack trace?

I see a call in the #to_s method which I think I will remove because it is a performance killer.

**Comment by gdknutel** [comment]
In calabash-ios, there is at least one call to `Instruments#version`. See https://github.com/calabash/calabash-ios/blob/5e8a568434ca8add2e82e0783137c85711b42166/calabash-cucumber/lib/calabash-cucumber/utils/simulator_accessibility.rb#L291.

**Comment by jmoody** [comment]
Yeah, that is dead code - it is never called.  Thanks for looking!

If fixed the `#version` call so it is much faster.

**Comment by gdknutel** [comment]
I should have just checked my stack trace. It was indeed the call from `to_s` that caused the trouble.

**Comment by jmoody** [comment]
Fixed in 1.5.5
